app/src/main/java/ch/epfl/sdp/peakar/camera/CameraUiView.java

```diff
@@ -83,11 +85,33 @@
     private Bitmap mountainMarkerVisible;
     private Bitmap mountainMarkerNotVisible;
 
-    //List that contains the POIPoints
-    private List<POIPoint> POIPoints;
     //Map that contains the labeled POIPoints
     private Map<POIPoint, Boolean> labeledPOIPoints;
 
+    private final SharedPreferences sharedPref;
+
+    private Boolean displayedToastMode;
+
+    private final SharedPreferences.OnSharedPreferenceChangeListener listenerPreferences =
+            (prefs, key) -> {
+                String displayMode = prefs.getString("display_pois_preference", "0");
+                boolean filterPOIs = prefs.getBoolean("filter_pois", true);
+
+                switch (displayMode){
+                    case "1":
```

Comment:
Use a meaningfull constant instead of `"1"` and `"2"` for the switch case. 
```suggestion
                    case DISPLAY_POI_IN_SIGHT:
```